### PR TITLE
#174797536 edit open travel request

### PR DESCRIPTION
**Pull request by valens10**
#### What does this PR do?
create edit open request endpoint
#### Description of Task to be completed?
- create a route for edit open request 
- create a controller for updating open request and send email to his/her LineManager
- add validations for edit open request router
- add tests for updating open request endpoint
#### How should this be manually tested?
- Clone the repo git clone `https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
- Switch your working directory `cd script-heroes-bn-backend`
- Checkout the branch git checkout `ft-update-open request-#174797536`
- Get your requests `http://localhost:3000/api/trips/`
- Update your request that is still open `http://localhost:3000/api/trips/open/requestId`
 - Expected body 
`{
"origin": 1,
"destination": 2,
"from": "2020-10-02",
"till": "2020-10-04"
}`

#### What are the relevant pivotal tracker stories?
[#174797536](https://www.pivotaltracker.com/story/show/174797536)

### CODE REVIEW

**Review by dushimeemma — APPROVED** [review]
Smels good to me but please work on tests

**Comment by valens10** [comment]
> Smels good to me but please work on tests

sure.

File: src/controllers/requests.js [hunk 1]
```diff
@@ -93,5 +101,56 @@ class RequestsController {
...
+    if (!request) {
+      return res.status(400).json({
+        error: res.__(notExistRequest)
+      });
+    }
+
+    if (request.userId !== user.id) {
+      return res.status(401).json({
+        error: res.__(notYourRequest)
+      });
```
Comment (elvisiraguha) [inline src/controllers/requests.js:127]: Could this be put into the middleware, let's keep only the actual business into the controller

**Comment by elvisiraguha** [comment]
Also add swagger documentation, and other languages translation.

File: locales/en.json [hunk 2]
```diff
@@ -197,9 +197,14 @@
...
 	"Trip deleted succesfully": "Trip deleted succesfully",
-	"No accomodation found": "No accomodation found"
+	"Trip updated succesfuly": "Trip updated succesfuly",
+	"Request updated succesfuly": "Request updated succesfuly",
+	"You have to provide at least one new information": "You have to provide at least one new information",
+	"You can not update a request which does not exist": "You can not update a request which does not exist",
+	"The request is updated successfully": "The request is updated successfully",
+	"No accomodation found": "No accomodation found",
+	"You can only update your request": "You can only update your request"
 }
```
Comment (Habinezajanvier) [inline locales/en.json]: Please add an empty line

**Review by elvisiraguha — APPROVED** [review]

**Comment by b0nbon1** [comment]
could we fix the tests?

**Review by b0nbon1 — APPROVED** [review]

**Pull request merged**